andela/merry-ah-backend
# #162727466 Users should be able to reset password

#### What does this PR do?
- This PR helps resets a users password when forgotten
#### Description of Task to be completed?
- The user should receive a mail on a valid email to confirm the password reset
- Once mail link is confirmed, the user should be able to reset password
- Password fields must match each other to complete a successful reset
#### How should this be manually tested?
- Run `npm i`
- Send a **POST** request to `/api/v1/auth/forgot-password`, pass a valid email address to the `email` field
- On successful request, copy the token from the response object and send a **PUT** request to `/api/v1/auth/forgot-password`, pass in the new password and confirm in fields `password` and `confirmPassword`
- Try signing in again on `/api/v1/auth/signin` to confirm the password reset
#### Any background context you want to provide?
- User must have signed up on the app before attempting to test this feature.
#### What are the relevant pivotal tracker stories?
[#162727466](https://www.pivotaltracker.com/story/show/162727466)
#### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2019-01-15 at 8 44 28 am" src="https://user-images.githubusercontent.com/42177767/51165870-dee71c00-18a1-11e9-95c7-0dbe26696c7d.png">
<img width="1440" alt="screen shot 2019-01-15 at 8 44 32 am" src="https://user-images.githubusercontent.com/42177767/51165871-df7fb280-18a1-11e9-9e01-7e25771f800f.png">

#### Questions:
N/A
